Test for importing pandas

In [261]:
import pandas as pd

In [262]:
url='https://covid.ourworldindata.org/data/owid-covid-data.csv'

In [263]:
df_covid_raw = pd.read_csv(url)
df_covid_raw

c:\Users\guido\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (33) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


iso_code continent     location        date  total_cases  new_cases  \
0           AFG      Asia  Afghanistan  2020-01-03          NaN        0.0   
1           AFG      Asia  Afghanistan  2020-01-04          NaN        0.0   
2           AFG      Asia  Afghanistan  2020-01-05          NaN        0.0   
3           AFG      Asia  Afghanistan  2020-01-06          NaN        0.0   
4           AFG      Asia  Afghanistan  2020-01-07          NaN        0.0   
...         ...       ...          ...         ...          ...        ...   
311511      ZWE    Africa     Zimbabwe  2023-05-13     264821.0        5.0   
311512      ZWE    Africa     Zimbabwe  2023-05-14     264841.0       20.0   
311513      ZWE    Africa     Zimbabwe  2023-05-15     264848.0        7.0   
311514      ZWE    Africa     Zimbabwe  2023-05-16     264848.0        0.0   
311515      ZWE    Africa     Zimbabwe  2023-05-17     264848.0        0.0   

        new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  \
0                      NaN           NaN         0.0                  NaN   
1                      NaN           NaN         0.0                  NaN   
2                      NaN           NaN         0.0                  NaN   
3                      NaN           NaN         0.0                  NaN   
4                      NaN           NaN         0.0                  NaN   
...                    ...           ...         ...                  ...   
311511              10.286        5690.0         1.0                0.429   
311512              10.714        5690.0         0.0                0.429   
311513              10.286        5690.0         0.0                0.286   
311514               9.429        5690.0         0.0                0.286   
311515               8.143        5690.0         0.0                0.143   

        ...  male_smokers  handwashing_facilities  hospital_beds_per_thousand  \
0       ...           NaN                  37.746                         0.5   
1       ...           NaN                  37.746                         0.5   
2       ...           NaN                  37.746                         0.5   
3       ...           NaN                  37.746                         0.5   
4       ...           NaN                  37.746                         0.5   
...     ...           ...                     ...                         ...   
311511  ...          30.7                  36.791                         1.7   
311512  ...          30.7                  36.791                         1.7   
311513  ...          30.7                  36.791                         1.7   
311514  ...          30.7                  36.791                         1.7   
311515  ...          30.7                  36.791                         1.7   

        life_expectancy  human_development_index  population  \
0                 64.83                    0.511  41128772.0   
1                 64.83                    0.511  41128772.0   
2                 64.83                    0.511  41128772.0   
3                 64.83                    0.511  41128772.0   
4                 64.83                    0.511  41128772.0   
...                 ...                      ...         ...   
311511            61.49                    0.571  16320539.0   
311512            61.49                    0.571  16320539.0   
311513            61.49                    0.571  16320539.0   
311514            61.49                    0.571  16320539.0   
311515            61.49                    0.571  16320539.0   

        excess_mortality_cumulative_absolute  excess_mortality_cumulative  \
0                                        NaN                          NaN   
1                                        NaN                          NaN   
2                                        NaN                          NaN   
3                                        NaN                          NaN   
4                 

In [264]:
df_covid_raw.shape

(311516, 67)

In [265]:
df_covid_raw.columns

Index(['iso_code', 'continent', 'location', 'date', 'total_cases', 'new_cases',
       'new_cases_smoothed', 'total_deaths', 'new_deaths',
       'new_deaths_smoothed', 'total_cases_per_million',
       'new_cases_per_million', 'new_cases_smoothed_per_million',
       'total_deaths_per_million', 'new_deaths_per_million',
       'new_deaths_smoothed_per_million', 'reproduction_rate', 'icu_patients',
       'icu_patients_per_million', 'hosp_patients',
       'hosp_patients_per_million', 'weekly_icu_admissions',
       'weekly_icu_admissions_per_million', 'weekly_hosp_admissions',
       'weekly_hosp_admissions_per_million', 'total_tests', 'new_tests',
       'total_tests_per_thousand', 'new_tests_per_thousand',
       'new_tests_smoothed', 'new_tests_smoothed_per_thousand',
       'positive_rate', 'tests_per_case', 'tests_units', 'total_vaccinations',
       'people_vaccinated', 'people_fully_vaccinated', 'total_boosters',
       'new_vaccinations', 'new_vaccinations_smoothed',
       't

In [266]:
continent_list = df_covid_raw['continent'].drop_duplicates().dropna()

In [267]:
df_total_cases_per_continent = df_covid_raw[['continent','location' ,'total_cases']].dropna().groupby(['continent', 'location']).max().groupby('continent').sum()
df_total_cases_per_continent

total_cases
continent                 
Africa          13087749.0
Asia           297229145.0
Europe         249319125.0
North America  124174879.0
Oceania         14073969.0
South America   68555165.0

In [268]:
df_total_vaccinations_per_continent = df_covid_raw[['continent','location' ,'total_vaccinations']].dropna().groupby(['continent', 'location']).max().groupby('continent').sum()
df_total_vaccinations_per_continent

total_vaccinations
continent                        
Africa               8.060550e+08
Asia                 9.007192e+09
Europe               1.552526e+09
North America        1.150121e+09
Oceania              8.100763e+07
South America        9.571019e+08

In [269]:
def compare_continents_covid_cases(df: pd.DataFrame, continents:list):
    
    total_cases = df_total_cases_per_continent.sum()

    continent_confront = df['continent'].isin(continents)

    df_filtered_per_location = df[continent_confront].loc[:,['continent', 'location','total_cases']].dropna().groupby(['continent', 'location'])['total_cases']
    df_filtered_per_continent = df[continent_confront].loc[:,['continent', 'location','total_cases']].dropna().groupby(['continent'])['total_cases']

    df_with_useful_values_per_location = df_filtered_per_location.agg(['min', 'max']).reset_index()
    df_with_useful_values_per_continent = df_filtered_per_continent.agg(['min', 'max'])
    
    df_with_useful_values_per_location['precentVsTotal'] = df_with_useful_values_per_location['max'].apply(lambda x: (x/total_cases))
    
    df_with_useful_values_per_continent['mean_max'] = df_with_useful_values_per_location.groupby(['continent'])['max'].mean()
    df_with_useful_values_per_continent['mean_min'] = df_with_useful_values_per_location.groupby(['continent'])['min'].mean()
    
    df_with_useful_values_per_continent['precentVsTotal'] = df_with_useful_values_per_location.groupby(['continent'])['max'].sum().apply(lambda x: (x/total_cases) * 100)

    print(df_with_useful_values_per_continent.reset_index())
    return(df_with_useful_values_per_location)


In [270]:
def compare_continents_vaccinations(df: pd.DataFrame, continents:list):
    
    total_vaccinations = df_covid_raw[['continent','location' ,'total_vaccinations']].dropna().groupby(['continent', 'location']).max().sum()

    continent_confront = df['continent'].isin(continents)

    df_filtered_per_location = df[continent_confront].loc[:,['continent', 'location','total_vaccinations']].dropna().groupby(['continent', 'location'])['total_vaccinations']
    df_filtered_per_continent = df[continent_confront].loc[:,['continent', 'location','total_vaccinations']].dropna().groupby(['continent'])['total_vaccinations']

    df_with_useful_values_per_location = df_filtered_per_location.agg(['min', 'max']).reset_index()
    df_with_useful_values_per_continent = df_filtered_per_continent.agg(['min', 'max'])
    
    df_with_useful_values_per_location['precentVsTotal'] = df_with_useful_values_per_location['max'].apply(lambda x: (x/total_vaccinations))
    
    df_with_useful_values_per_continent['mean_max'] = df_with_useful_values_per_location.groupby(['continent'])['max'].mean()
    df_with_useful_values_per_continent['mean_min'] = df_with_useful_values_per_location.groupby(['continent'])['min'].mean()

    df_with_useful_values_per_continent['precentVsTotal'] = df_with_useful_values_per_location.groupby(['continent'])['max'].sum().apply(lambda x: (x/total_vaccinations) * 100)

    print(df_with_useful_values_per_continent.reset_index())

    return(df_with_useful_values_per_location)

In [271]:
compare_continents_covid_cases(df_covid_raw, ['Europe', 'South America', 'Oceania'])

continent
Europe           249319125.0
Oceania           14073969.0
South America     68555165.0
Name: max, dtype: float64
       continent  min         max      mean_max  mean_min  precentVsTotal
0         Europe  1.0  38989402.0  4.888610e+06  1.274510       32.529502
1        Oceania  1.0  11299954.0  5.864154e+05  3.166667        1.836278
2  South America  1.0  37511921.0  4.896798e+06  3.142857        8.944622


continent   location  min        max  precentVsTotal
0          Europe    Albania  2.0   334090.0        0.000436
1          Europe    Andorra  1.0    48015.0        0.000063
2          Europe    Austria  1.0  6072229.0        0.007923
3          Europe    Belarus  1.0   994037.0        0.001297
4          Europe    Belgium  1.0  4797108.0        0.006259
..            ...        ...  ...        ...             ...
84  South America   Paraguay  1.0   735759.0        0.000960
85  South America       Peru  9.0  4505220.0        0.005878
86  South America   Suriname  1.0    82495.0        0.000108
87  South America    Uruguay  4.0  1037893.0        0.001354
88  South America  Venezuela  6.0   552695.0        0.000721

[89 rows x 5 columns]

In [272]:
compare_continents_vaccinations(df_covid_raw, ['Europe', 'South America', 'Oceania'])

continent
Europe           1.552526e+09
Oceania          8.100763e+07
South America    9.571019e+08
Name: max, dtype: float64
       continent  min          max      mean_max       mean_min  \
0         Europe  0.0  192221468.0  2.875048e+07  105959.111111   
1        Oceania  0.0   63681652.0  4.500424e+06    1170.333333   
2  South America  0.0  486436436.0  7.362322e+07    2083.923077   

   precentVsTotal  
0       11.454373  
1        0.597666  
2        7.061396  


continent   location      min         max  precentVsTotal
0          Europe    Albania      0.0   3080679.0        0.000227
1          Europe    Andorra    576.0    157038.0        0.000012
2          Europe    Austria  31789.0  20464490.0        0.001510
3          Europe    Belarus      0.0  19287182.0        0.001423
4          Europe    Belgium    340.0  29633317.0        0.002186
..            ...        ...      ...         ...             ...
80  South America   Paraguay    200.0   9745545.0        0.000719
81  South America       Peru   3859.0  89481431.0        0.006602
82  South America   Suriname      0.0    505699.0        0.000037
83  South America    Uruguay    383.0   9021252.0        0.000666
84  South America  Venezuela      0.0  37860994.0        0.002793

[85 rows x 5 columns]

L'europa ha il 32% dei casi totali registrati e l'11% dei delle vaccinazioni totali. Quindi ha il numero totale di casi e vaccinazioni più alti rispetto ai 3 continenti presi in analisi.
Il continente con meno casi e vaccinazioni è l'oceania che vanta circa l'2% dei casi totali di covid e meno dell'1% delle vaccinazioni totali
Il sud-america ha il circa il 9% dei casi totali e il 7% delle vaccinazioni totali, una nota riguardo il sud America è che la media dei massimi dei casi totali per locazione è maggiore rispetto a quella dell'europa, mentre la media massima di vaccinazioni per locazione, è molto più alta di quella europea.